tgui/packages/tgui/components/Dropdown.tsx

```diff
@@ -234,6 +237,55 @@ export class Dropdown extends Component<DropdownProps, DropdownState> {
     }
   }
 
+  getOptionValue(option): string {
+    return typeof option === 'string' ? option : option.value;
+  }
+
+  getOptionsValues(): string[] {
+    const { options = [] } = this.props;
+
+    return options.map((option) => {
+      return this.getOptionValue(option);
+    });
+  }
+
+  getSelectedIndex(): number {
+    const selected = this.state.selected || this.props.selected;
+    const { options = [] } = this.props;
+
+    return options.findIndex((option) => {
+      return this.getOptionValue(option) === selected;
+    });
+  }
+
+  toPrevious(): void {
+    const selectedIndex = this.getSelectedIndex();
+
+    if (selectedIndex < 0) {
+      return;
+    }
```

Comment:
If nothing is selected (`selectedIndex  === -1`), then the buttons are clickable, but do nothing?

Reply:
Yes

Reply:
That's not a great UX, because no feedback will be given. It's better to switch to last option in toPrevious and to first in toNext if there was a placeholder instead of an option.

Reply:
In normal case this will never call, selected value always will be in options range, but it can be bug (if selected value not assign or out options range). I think in bug case better nothing do

Reply:
Dropdowns without options should probably be considered bugs IMO, I don't think we have any in the codebase

Reply:
It was about a case without an initial selected option, which isn't requirement. Like this:
```js
// Values
const selected = null; // Initial value, changes after selection.
const options = ["first", "second"];
...
// Render
<Dropdown
  selected={selected}
  options={options}
  displayText={selected || 'Choose an option'}
/>
```
This code would produce a valid typical Dropdown with a placeholder text if no option selected. Arrow buttons would look clickable, but would do nothing on click, which isn't a great UX.